### PR TITLE
Enable UnusedVariable error prone check

### PR DESCRIPTION
**Pull request by brandboat**
#4188

Enable `UnusedVariable`

### CODE REVIEW

**Review by Lewuathe — COMMENTED** [review]
@brandboat Could you look into `OrcPageSource` as well to remove (or suppress) unused variable?
 
```
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.0:compile (default-compile) on project presto-hive: Compilation failure: Compilation failure: 
Error:  /home/runner/work/presto/presto/presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSource.java:[62,33] [UnusedVariable] The field 'log' is never read.
Error:      (see https://errorprone.info/bugpattern/UnusedVariable)
Error:    Did you mean to remove this line or 'static { Logger.get(OrcPageSource.class); }'?
Error:  /home/runner/work/presto/presto/presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcFileWriterFactory.java:[190,37] [UnusedVariable] The local variable 'rootType' is never read.
Error:      (see https://errorprone.info/bugpattern/UnusedVariable)
Error:    Did you mean to remove this line?
Error:  -> [Help 1]
```

**Comment by brandboat** [comment]
> @brandboat Could you look into OrcPageSource as well to remove (or suppress) unused variable?

Sure ! It would be great if anyone could make a quick review on this PR since code change a lot everyday while unused variable is very common.

File: presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQueryClient.java [hunk 1]
```diff
@@ -62,6 +62,7 @@
     }
 
     // return empty if no filters are used
+    @SuppressWarnings("unused")
```
Comment (findepi) [inline presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQueryClient.java:65]: We should remove the method instead. No value in keeping it.

File: presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQueryConnectorModule.java [hunk 2]
```diff
@@ -32,6 +32,7 @@
 public class BigQueryConnectorModule
         implements Module
 {
+    @SuppressWarnings("unused")
```
Comment (findepi) [inline presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQueryConnectorModule.java:35]: if the field is unused why not remove it?

File: presto-druid/src/main/java/io/prestosql/plugin/druid/DruidErrorCode.java [hunk 3]
```diff
@@ -25,6 +25,7 @@
     DRUID_DDL_NOT_SUPPORTED(0, EXTERNAL),
     DRUID_DML_NOT_SUPPORTED(1, EXTERNAL);
 
+    @SuppressWarnings("unused")
```
Comment (findepi) [inline presto-druid/src/main/java/io/prestosql/plugin/druid/DruidErrorCode.java:28]: @samarthjain why is it unused?

**Comment by findepi** [comment]
> Sure ! It would be great if anyone could make a quick review on this PR since code change a lot everyday while unused variable is very common.

Good point.
@brandboat to make your and reviewer's job easier, i would suggest to split this pull request into multiple separate pull requests (not commits), so that we can merge incrementally.
for example "Fix unused variables in presto-bigquery",  "Fix unused variables in presto-cassandra",  "Fix unused variables in presto-hive", etc.
We will then enable the errorprone check at the end.

Also, we should avoid marking the unused places with `@SuppressWarning`. The point is rather to remove them, if possible.
By having it in smaller PRs, it will be easier to process.

Does it make sense?

**Comment by brandboat** [comment]
> @brandboat to make your and reviewer's job easier, i would suggest to split this pull request into multiple separate pull requests (not commits), so that we can merge incrementally.
for example "Fix unused variables in presto-bigquery", "Fix unused variables in presto-cassandra", "Fix unused variables in presto-hive", etc.
We will then enable the errorprone check at the end.

@findepi Thanks for your comment !
Agree, there are too many code changes in this PR.

> Also, we should avoid marking the unused places with @SuppressWarning. The point is rather to remove them, if possible.
By having it in smaller PRs, it will be easier to process.

Yes, we should not leave useless variable. Adding @SuppressWarning in this PR is when refactoring out unused variable may change lots of files and make this PR hard to deal with. But if we split this PR by package or in smaller PR, we could solve all unused variables piece by piece.

I will open other PRs later.

**Comment by findepi** [comment]
> Yes, we should not leave useless variable. Adding @SuppressWarning in this PR is when refactoring out unused variable may change lots of files and make this PR hard to deal with.

That's an important observation and thanks for taking care of reviewability of the change.
Note that proposed split "per module" is just a proposal.
in particular, if there are many changes in given module, it may still be hard to read.

**Comment by brandboat** [comment]
Close this PR and open another new PR in #5987.

**Pull request closed without merging**